### ISSUE TITLE
duplicate closing tbody tag produces false positive

### ISSUE THREAD

**Issue by django-dou7** [validator/validator#2142]
### Summary

A stray closing `tbody` tag within a table doesn't produce an error or warning.

### URL for document that reproduces the problem

n/a

### HTML/CSS/SVG snippet that reproduces the problem

```html
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>HTML Elements</title>
</head>
<body>
    <table>
        <caption>Fictional Characters</caption>
        <thead>
            <tr>
                <th>Character</th>
                <th>Series</th>
                <th>Age</th>
            </tr>
        </thead>
        <tbody>
            <tr>
                <td>Batman</td>
                <td>Batman</td>
                <td>35</td>
            </tr>
            <tr>
                <td>Spider-Man</td>
                <td>Spider-Man</td>
                <td>25</td>
            </tr>
        </tbody>
            <tr>
                <td>Joker</td>
                <td>Batman</td>
                <td>???</td>
            </tr>
        </tbody>
    </table>
</body>
</html>
```

### Checker error messages

```shell
n/a
```

### Steps to reproduce

n/a

### Expected behavior

An unclosed tag should produce an error

### Actual behavior

<img width="803" height="722" alt="Image" src="https://github.com/user-attachments/assets/fc215381-6fcb-47e7-b505-85e1b0cce684" />
